contracts/PoolManager.sol

```diff
@@ -318,8 +329,13 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC1155, IERC1155Rec
 
     /// @inheritdoc IPoolManager
     function take(Currency currency, address to, uint256 amount) external override noDelegateCall onlyByLocker {
-        _accountDelta(currency, amount.toInt128());
-        reservesOf[currency] -= amount;
+        Currency accountCurrency = (currency == WRAPPED_NATIVE) ? CurrencyLibrary.NATIVE : currency;
+
+        _accountDelta(accountCurrency, amount.toInt128());
+        reservesOf[accountCurrency] -= amount;
+
+        if (currency == WRAPPED_NATIVE) WETH_CONTRACT.deposit{value: amount}();
```

Comment:
can we instead abstract this into `CurrencyLibrary`? ie have `Currency.transfer(to, amount)` do wrap + transfer if WETH given as input token

Reply:
Only thought would be then to make Currency not a library and just a contract.. then we could pass it through to the inherited contract. Not great but can't think of anything else